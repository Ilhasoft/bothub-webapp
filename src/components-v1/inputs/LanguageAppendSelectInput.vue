<template>
  <bh-dropdown class="example-txt-w-highlighted-entities__input__language">
    <span
      slot="trigger"
      class="clickable">
      <bh-language-flag
        v-if="val"
        :language="val" />
      <bh-icon
        v-else
        value="earth" />
      <bh-icon value="menu-down" />
    </span>
    <bh-dropdown-item
      v-for="(verbose, language) in LANGUAGES"
      :key="language"
      @click="setVal(language)">
      <bh-language-flag
        :language="language"
        size="small" />
      <span>{{ verbose }}</span>
    </bh-dropdown-item>
  </bh-dropdown>
</template>

<script>
import { LANGUAGES } from '@/utils';


export default {
  name: 'LanguageAppendSelectInput',
  props: {
    value: {
      type: String,
      default: null,
    },
  },
  data() {
    return {
      val: this.value,
      LANGUAGES,
    };
  },
  watch: {
    value(value) {
      this.val = value;
    },
    val(value) {
      this.$emit('input', value);
    },
  },
  methods: {
    setVal(value) {
      this.val = value;
    },
  },
};
</script>
