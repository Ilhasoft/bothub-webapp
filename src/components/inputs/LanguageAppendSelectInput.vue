<template>
  <b-dropdown
    :scrollable="true"
    :position="dropdownDirection"
    :max-height="175"
    class="language-append-select-input">
    <span
      slot="trigger"
      class="clickable language-append-select-input__language-select">
      <language-badge
        v-if="val"
        :language="val" />
      <b-icon
        v-else
        class="language-icon language-icon__menu__no-val"
        size="is-small"
        icon="earth" />
      <b-icon
        class="clickable align language-icon language-icon__menu"
        size="is-small"
        icon="menu-down" />
    </span>
    <b-dropdown-item
      v-for="(verbose, language) in languageList"
      :key="language"
      @click="setVal(language)">
      <span>{{ verbose }}</span>
    </b-dropdown-item>
  </b-dropdown>
</template>

<script>
import { LANGUAGES, languageListToDict } from '@/utils';
import LanguageBadge from '@/components/shared/LanguageBadge';


export default {
  name: 'LanguageAppendSelectInput',
  components: {
    LanguageBadge,
  },
  props: {
    value: {
      type: String,
      default: null,
    },
    languages: {
      type: Array,
      default: null,
    },
    dropdownDirection: {
      type: String,
      default: 'is-bottom-left',
    },
  },
  data() {
    return {
      val: this.value,
      LANGUAGES,
    };
  },
  computed: {
    languageList() {
      if (!this.languages) return LANGUAGES;
      return languageListToDict(this.languages);
    },
  },
  watch: {
    value(value) {
      this.val = value;
    },
    val(value) {
      this.$emit('input', value);
    },
  },
  methods: {
    setVal(value) {
      this.val = value;
    },
  },
};
</script>

<style lang="scss">
.language-append-select-input {

  min-height: 30px;

  &__language-select {
    align-items: center;
  }
}

.language-icon {
  color: black !important;
  pointer-events: all !important;

  &__menu {
    width: 1rem !important;

    &__no-val {
      left: -1.5rem;
    }
  }
}
</style>
