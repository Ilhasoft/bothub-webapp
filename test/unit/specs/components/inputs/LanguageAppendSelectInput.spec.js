import BH from 'bh';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import LanguageAppendSelectInput from '@/components/inputs/LanguageAppendSelectInput';


const localVue = createLocalVue();
localVue.use(BH);

describe('LanguageAppendSelectInput.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(LanguageAppendSelectInput, { localVue });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('click in en language', () => {
    beforeEach(() => {
      wrapper.vm.setVal('en');
    });

    test('input event emitted', () => {
      expect(wrapper.emitted('input')).toBeDefined();
    });
  });

  describe('value prop change to pt', () => {
    beforeEach(() => {
      wrapper.setProps({ value: 'pt' });
    });

    test('val is pt', () => {
      expect(wrapper.vm.val).toBe('pt');
    });
  });
});
