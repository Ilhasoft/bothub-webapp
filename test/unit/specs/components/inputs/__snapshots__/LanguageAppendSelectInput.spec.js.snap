// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageAppendSelectInput.vue renders correctly 1`] = `
<b-dropdown-stub scrollable="true" maxheight="175" position="is-bottom-left" mobilemodal="true" animation="fade" closeonclick="true" canclose="true" class="language-append-select-input"><span class="clickable language-append-select-input__language-select"><b-icon-stub icon="earth" size="is-small" class="language-icon language-icon__menu__no-val"></b-icon-stub> <b-icon-stub icon="menu-down" size="is-small" class="clickable align language-icon language-icon__menu"></b-icon-stub></span>
  <b-dropdown-item-stub focusable="true" ariarole=""><span>English</span></b-dropdown-item-stub>
  <b-dropdown-item-stub focusable="true" ariarole=""><span>Portuguese</span></b-dropdown-item-stub>
</b-dropdown-stub>
`;
